### ISSUE TITLE
option dependencies on boolean true doesn not seem to always work

### ISSUE THREAD

**Issue by ewaldgh** [json-editor/json-editor#1126]
### General information

- json-editor version: latest

### Expected behavior
ENABLE_PTT ("Enable Push-To-Talk IP Radio Communications") boolean (checkbox) correctly displays/hides it's dependent properties.
ENABLE_CALLING ("Enable VOIP") is expected to behave the same way and display/hide it's dependent properties

### Actual behavior
However, ENABLE_CALLING ("Enable VOIP") does not display/hide its dependent property.    I tried renaming the property, but no change.
When I look in the Firefox/Chrome dev tools, it seems that the "erronuous" div object does not have the "display: [block|none]" style.
_BAD_: `<div data-schematype="boolean" data-schemapath="root.ENABLE_CALLING" class="col-md-12">`
_GOOD_: `<div data-schematype="string" data-schemapath="root.PTT_URL" class="col-md-12" style="display: block;">`

### Steps to reproduce the behavior

[Direct link to example](https://json-editor.github.io/json-editor/?data=N4Ig9gDgLglmB2BnEAuUMDGCA2MBGqIAZglAIYDuApomALZUCsIANOHgFZUZQD62ZAJ5gArlELwwAJzplsrEIgwALKrNShYUbFUIBhBERgBzEVLKwECqIIi6U7LjwVSqARxExXAE1QBtYjlsXm8qKG4+KngyPB1fNgBRADkAQQAhABkE3j0UjIyASSSAcQUANzAYCAA6XAwoxCpeAGsqQQUANQB5Ar0EjJSATQSAJV4UgAUJ3lSAWQTOnr6B4bHJ6YBpBMHF3v6h0fGp3gBlBL0RhIAVBWT0rN4Jq5u2J6vHlJOTgHUukYARBRvXgAVRGGSBz0eK14swKXwS/1hCS+KWKKMh71mKQAGrwCv8HnouiCkldRpjTlcUlcQSccgAJc4bXhXArzEYgAC6bAgUkgVCksBoGkC2GCoXCPF4URicVFNjshDwYDAOjI8AUJBkFkIKm4zRVAA9rDBtPYQAlorEqAACABiQVt/zCETg8FtswQZukMHgxltGu8tpSOiFfuMClCRDIImw4hQUCkIioAF82DHxSFXTx3SEYIhyPB6grbBb4CI6HhBabzYR/gWi/VbSJGsHtQ6nS6pZYPV74D6pBHbQAKOgASgUdD9MDoldQAEY2LIjbP5ygABxsaOx+OL6qMABMABZ05bUplsrl8kVSmgQIqLSq1VQNVrpLIEyB9RhDWATWwWg6IQVpyna3QFBMUZUDGcYJpmjRnhUVS1JgDRNK07T3o+hCFkO/oKJAvbIPeoR2PAoTFjAIr3ncl45HkhQlKgSYpqmZ5ARaEETLaGRoUgdobG0IBnnRDxvKWSoOM+6qahmH66g4P5/gBD5msBDigTatoTK2ygALRXGAhlyM0tqQbaIxkN4cC2gYdBzgOGAWO6yDbjBu7wXIiGvFCYIQthZa4UmEa1hpIAdJU9QZEIgq2v5Livt4XTwNgWGsVQbBEa5opkVElEYNRJGgGJ2QSYmyZpme07wBkUTGFAyioAADMuZBGnV/qNagADMh5nhBywHGsxxzAsgVSYoIUEYB6lcVFVAxYIcUpBAEB1C5CC2kkZAMIR0A5aRVDkQVRWiqVjzPCxlXse5sF7g4phkFI3jKLqA1LPsqxHJs2ySRaeGhbNdYOJFaFLSta0bb2tpCe0WUHQgxUgHlFFRIVNElRe4lXRVbG3RFn0rIc6ynOclw3BNAPTZGwPhWD0WxVIIZQ5gm0eic3CuOICPEblx35ejZ20djZW4xlBM1Z1DVNSgrUgCu0vdSgfVnsCEyfD8fyAlTwX4bTxAKV+EBkIgiAUNI8RqSDEULRDzMm2bFsvftfNHSdQuY+e9xi5TEvVX6Suy/Liv1crqu+e8EwwnCCJIvMqLoic/3KqqsnvjqX7KcaYUWlpOg6QIgi2mARC2gwZtkMYNC2o1Fjl0ItpvWUdrVlEtqNPAUC2hQygwAXpdELg8C6Lzh2gKjp1exd5X+3dnmoAhaaR7CuL4oSV4kmSFK6w4frhNXUi54QsztWudC2jA3gF1gIhdzWbBROufiHiwx48uAiNIPzHtUdPouXT9jdR+FY6C8E4iRAIh5bT6VtB0QUMAABeVg2DHm5GeHccFUCnhXicaktJ6R6CZHoFkbIOQpz3vfQ+x8HAnHIFAVstplK11nHFMciBJwgOfj1ZqvDWoADY+HNQ/tlJGP9BZ/2RjPcWwCQBPzARA/wIAeF8IUIIvhMC4EIOQZqLkGCPJYJVkI9iZ5EDKDABQGUUh+RSBIiAfegoyC5hQQ+VQe1pKqigHhMgEA0HsSAA=)

Apologies if this is a user error, it's my first experience with this superb editor.

**Comment by ewaldgh** [comment]
Got it !  The name of the dependent property can not contain a dot !

**Comment by ewaldgh** [comment]
I am reopening the issues because I could not find where in the spec it says that keys can not contain dots, just that they have to be "strings".  The schema also past several online validations.   It's not a big issue, but if it's a limitation of the implementation, maybe it would be good to mention in documentation.  My problem is solved though, so consider this low priorty.
Thanks in advance.

**Comment by germanbisurgi** [comment]
This [example]("title": "b") will clear that question.
One of the properties (c) can be dependent on a sibling property (a) property (a.b)
The Dot character is used to access nested properties in objects

**Comment by ewaldgh** [comment]
Indeed, I imagined something like that.  It would be good though to point out this restriction as per JSON and JSON schema standards a key can be any string, with the restriction that some characters need escaping.  Before posting this issue I checked many places to validate that I had used a valid key name, but could not find any indication otherwise. For example: [discussion on stackexchange on allowed keys](https://stackoverflow.com/questions/8676011/which-characters-are-valid-invalid-in-a-json-key-name).    
Maybe the syntax for nested dependencies could become "a": { "b": true } as in my (simple) mind the impact of someone having to change JSON keys (e.g. in firmware) is bigger than having to change a schema.  Just a tought...
If you point me to concerned code, I am happy to help implement an alternative.
